events.go

```diff
@@ -240,9 +232,9 @@ func (s *Session) handleNodeUp(ip net.IP, port int, waitForBinary bool) {
 
 	host := s.ring.getHost(ip)
 	if host != nil {
-		if s.cfg.IgnorePeerAddr && host.Peer().Equal(ip) {
+		if s.cfg.IgnorePeerAddr && host.ConnectAddress().Equal(ip) {
```

Comment:
looking at it this code looks wrong, should it be `s.cfg.IgnorePeerAddr && !host.ConnectAddress().Equal(ip)`?